thrift0.13/src/main/java/com/linecorp/armeria/common/thrift/DefaultThriftProtocolFactoryProvider.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.thrift;
+
+import java.util.Set;
+
+import com.google.common.collect.ImmutableSet;
+
+/**
+ * Default registered {@link ThriftProtocolFactoryProvider}.
+ * It is not overridable but you may provide and register another implementation.
+ */
+public final class DefaultThriftProtocolFactoryProvider extends ThriftProtocolFactoryProvider {
+    @Override
+    protected Set<Entry> entries() {
+        return ImmutableSet.of(
+                new Entry(
+                        ThriftSerializationFormats.BINARY, ThriftProtocolFactories.BINARY),
+                new Entry(
+                        ThriftSerializationFormats.COMPACT, ThriftProtocolFactories.COMPACT),
+                new Entry(
+                        ThriftSerializationFormats.JSON, ThriftProtocolFactories.JSON),
+                new Entry(
+                        ThriftSerializationFormats.TEXT, ThriftProtocolFactories.TEXT),
+                new Entry(
+                        ThriftSerializationFormats.TEXT_NAMED_ENUM, ThriftProtocolFactories.TEXT_NAMED_ENUM)
+        );
```

Comment:
How about making this static final field?

Reply:
as I understand it, it's read only once in the JVM lifecycle, so it doesn't make sense to cache it.

Reply:
I thought this method could be called by users because this is a public API. Anyway, It makes sense to not cache.

Reply:
This is a protected method (not public API)

Reply:
Oops... I misused the term. 😅 What I meant was a user-accessible API.